### DISCUSSION TITLE
PyG macOS GPU

### DISCUSSION THREAD

**Discussion by ghost** [pyg-team/pytorch_geometric#7224] [Q&A]
Hi all! 
I’m just getting started learning about graph machine learning, and have set up a project using PyG.
I’m super new to this, but I was just wondering does PyG currently have support for utilizing the GPU on M1/2 chips? I know PyTorch has support for using the GPU through Metal but I can’t find any information about this for PyG.
If there isn’t support, are there plans to add support?

-Adrian

**Comment by rusty1s** [comment]
The honest answer is no, not all ops we use from PyTorch do support mps yet. You can either use `device==cpu` or try to run it with `mps` enabled and see where it might crash.
